api/src/shared/infrastructure/repositories/challenge-repository.js

```diff
@@ -122,38 +120,29 @@ export async function findValidatedBySkills(skills, locale) {
 }
 
 export async function findActiveFlashCompatible({
-  date,
   locale,
   successProbabilityThreshold = config.features.successProbabilityThreshold,
   accessibilityAdjustmentNeeded = false,
-  complementaryCertificationKey,
+  version,
   dependencies = {
     getInstance,
   },
 } = {}) {
   _assertLocaleIsDefined(locale);
-
-  const hasCoreScope =
-    !complementaryCertificationKey || complementaryCertificationKey === ComplementaryCertificationKeys.CLEA;
-
-  const cacheKey = `findActiveFlashCompatible({ scope: ${hasCoreScope ? Frameworks.CORE : complementaryCertificationKey}, locale: ${locale}, accessibilityAdjustmentNeeded: ${accessibilityAdjustmentNeeded}})`;
-
+  const cacheKey = `findActiveFlashCompatible({ versionId: ${version?.id}, locale: ${locale}, accessibilityAdjustmentNeeded: ${accessibilityAdjustmentNeeded} })`;
   let challengeDtos;
 
-  if (hasCoreScope) {
-    challengeDtos = await _findChallengesForCoreCertification({
-      locale,
-      accessibilityAdjustmentNeeded,
+  if (version && version.scope !== Frameworks.CORE) {
+    challengeDtos = await _findValidChallengesForComplementaryCertification({
       cacheKey,
+      versionId: version.id,
       dependencies,
     });
   } else {
-    const version = getVersionNumber(date);
-
-    challengeDtos = await _findValidChallengesForComplementaryCertification({
-      complementaryCertificationKey,
+    challengeDtos = await _findChallengesForCoreCertification({
+      locale,
+      accessibilityAdjustmentNeeded,
```

Comment:
Je simplifierai en inversant le if/else (et que le CORE est au final notre cas nominal) + le petit cote [clean code](https://csharpdeveloper.wordpress.com/2016/01/20/roberts-rules-of-coders-8-avoid-negative-conditionals-and-method-names/) sur les conditions negatives

```suggestion
  if (version?.scope === Frameworks.CORE) {
  
  challengeDtos = await _findChallengesForCoreCertification({
      locale,
      accessibilityAdjustmentNeeded,
} else {
     challengeDtos = await _findValidChallengesForComplementaryCertification({
cacheKey,
      versionId: version.id,
dependencies,
});
```